### PR TITLE
Add johnsnowlabs flavor docs

### PR DESCRIPTION
**Pull request by C-K-Loan**
Signed-off-by: Christian Kasim Loan <christian.kasim.loan@gmail.com>
@Ben-Epstein @harupy 

## Related Issues/PRs
https://github.com/mlflow/mlflow/issues/8552

## What changes are proposed in this pull request?
Docs for johnsnowlabs flavor.

WIP, this adds the documentation section to the  `python api`  for johnsnowlabs flavor.
I still need to prepare a colab/databricks tutorial notebook and add something to `models.rst` 

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
@mlflow-automation autoformat

**Comment by harupy** [comment]
thanks for the pr!

**Comment by C-K-Loan** [comment]
added entry to `models.rst`

only things left is 
1) add notebooks for databrricks + colab
2) add entry to the examples readme https://github.com/mlflow/mlflow/blob/master/examples/README.md 


btw please check this out https://github.com/mlflow/mlflow/pull/8687

File: mlflow/johnsnowlabs.py [hunk 1]
```diff
@@ -2,20 +2,27 @@
 The ``mlflow.johnsnowlabs`` module provides an API for logging and loading Spark NLP and NLU models.
 This module exports the following flavors:
 
-The flavor gives you access to `20.000+ state-of-the-art enterprise NLP models in 200+ languages
+To get started quickly, we recommend checking out
+1. The Google Colab tutorial Notebook TODO
+2. The Databricks tutorial Notebook TODO
```
Comment (harupy) [inline mlflow/johnsnowlabs.py:7]: We can add these notebook in a follow-up, don't need to make this PR larger.

**Comment by harupy** [comment]
@mlflow-automation autoformat

**Comment by harupy** [comment]
@C-K-Loan Please let us know the docs and API reference look ok. Once confirmed, we can cherry-picks the updates into MLflow 2.4.1's documentation :)

**Comment by C-K-Loan** [comment]
@harupy  sounds good then let's merge this, I will make a follow-up PR for the notebooks.

It's a bit tricky to make the notebooks right now anyways since we broke the flavor in the latest release :/ 

Please checkout this, right now the flavor is broken until we merge this and make a new mlflow release I think 
https://github.com/mlflow/mlflow/pull/8687

**Comment by harupy** [comment]
> Please checkout this, right now the flavor is broken until we merge this and make a new mlflow release I think


We released MLflow 2.4.1. This should be no longer an issue, right?

**Comment by C-K-Loan** [comment]
@harupy  No, the bug was introduced with MLflow 2.4.1, we need another release or at  least an RC.

the serve comand is broken, requirements.txt has `mlflow==2.4` in it's requirements which causes the venv to install mlflow 2.4.0 and give `module not found mlflow.johnsnowlabs`  since that version does not have the flavor


It more described and fixed in https://github.com/mlflow/mlflow/pull/8687

**Comment by C-K-Loan** [comment]
You can also see this problem in this notebook. 
Its running mlflow 2.4.1 but the venv rurns 2.4.0
image

**Comment by harupy** [comment]
The version pinning issue is a separate issue. Merging this PR.

**Pull request merged**